src/Domain/Creators/TimeDependentOptions/ExpansionMap.hpp

```diff
@@ -0,0 +1,107 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <optional>
+#include <string>
+#include <variant>
+
+#include "DataStructures/DataVector.hpp"
+#include "Domain/Creators/TimeDependentOptions/FromVolumeFile.hpp"
+#include "Options/Auto.hpp"
+#include "Options/Context.hpp"
+#include "Options/String.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace domain::creators::time_dependent_options {
+/*!
+ * \brief Class to be used as an option for initializing expansion map
+ * coefficients.
+ */
+struct ExpansionMapOptions {
+  using type = Options::Auto<ExpansionMapOptions, Options::AutoLabel::None>;
+  static std::string name() { return "ExpansionMap"; }
+  static constexpr Options::String help = {
+      "Options for a time-dependent expansion of the coordinates. Specify "
+      "'None' to not use this map."};
+
+  struct InitialValues {
+    using type =
+        std::variant<std::array<double, 3>, FromVolumeFile<names::Expansion>>;
+    static constexpr Options::String help = {
+        "Initial values for the expansion map, its velocity and "
+        "acceleration."};
+  };
+
+  struct InitialValuesOuterBoundary {
+    using type =
+        std::variant<std::array<double, 3>, FromVolumeFile<names::Expansion>>;
+    static constexpr Options::String help = {
+        "Initial values for the expansion map, its velocity and "
+        "acceleration at the outer boundary. Unless you are starting from a "
+        "checkpoint or continuing an evolution, this option should likely be "
+        "[1.0, 0.0, 0.0] at the start of an evolution"};
+  };
+
+  struct DecayTimescaleOuterBoundary {
+    using type = Options::Auto<double>;
+    static constexpr Options::String help = {
+        "A timescale for how fast the outer boundary expansion approaches its "
+        "asymptotic value. Can optionally specify 'Auto' when reading the "
+        "initial values 'FromVolumeFile' to use the decay timescale from the "
+        "function of time in the volume file. Cannot specify 'Auto' when "
+        "initial values are specified directly."};
+  };
+
+  struct DecayTimescale {
+    using type = Options::Auto<double, Options::AutoLabel::None>;
```

Comment:
What should this option be when continuing an evolution from volume data? Generally, there are lots of choices to make for all these options, but when I continue an evolution from volume data I just want to specify "FromVolumeFile" for the whole thing, don't I?

Reply:
It should be `None`. Yeah the way it is now probably isn't ideal. Yes just specifying "FromVolumeFile" would be nice, except that'd basically mean moving the "FromVolumeFile" part up a level and adding a separate "ExpansionMap" class for it.

I could change this from `None` to `Auto` so it's at least the same as the other options so when you are specifying that you want to continue from volume data, you just say `Auto` for everything else.

Reply:
@nilsvu Went ahead and made this `Auto` and just added a note in the help string about it